### PR TITLE
Update object compliance of OrderAttribute

### LINKED ISSUE #145: Compliance of OrderAttribute
- [x] BH.oM.Reflection.OrderAttribute

IObject needed and change of namespace.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
Fixes #145

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@BHoMBot check serialisation

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance
@BHoMBot check core
@BHoMBot check null-handling

File: EnergyPlus_oM/OrderAttribute.cs [hunk 1]
```diff
@@ -26,11 +26,12 @@
...
 using System.Text;
 using System.Threading.Tasks;
+using BH.oM.Base;
 
-namespace BH.oM.Reflection
+namespace BH.oM.Base.Attributes
 {
     [AttributeUsage(AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
-    public class OrderAttribute : Attribute
+    public class OrderAttribute : Attribute, IObject
```
Comment (IsakNaslundBh) [inline EnergyPlus_oM/OrderAttribute.cs:34]: WOnder if this needs verisoning support? Might not be in the list, as it was not an IObject before.

Also, not sure if the internal structure of this class makes sense? The constructor sets a private field, but then there is a public proeprty, that has 0 reference to the set field?

Not scope of this PR, but just highlighting as seen.

**Review by IsakNaslundBh — COMMENTED** [review]
Out of scope comment below.

Can approve, but also rather have @tg359 do that as I have 0 knowledge of this codebase.

File: EnergyPlus_oM/OrderAttribute.cs (hunk 1 above)
Comment (FraserGreenroyd) [reply EnergyPlus_oM/OrderAttribute.cs:34]: I do agree @IsakNaslundBh but am going to leave for now as out of scope of the immediate problem. This toolkit is alpha only so versioning isn't needed, but @jamesramsden-bh could you review this at some point to fix or take the toolkit out of alpha?

I'm going to get @EKAdebo to review this PR using a test script to verify we're ok to merge this as part of the work for the `IObject` compliance.

**Review by EKAdebo — APPROVED** [review]
This was updated as expected.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: ready-to-merge

**Pull request merged**